### PR TITLE
Verify MFD fix in non branding

### PR DESCRIPTION
**Pull request by MaryGao**
merged my change and qiaoqiao's change and re-generate non-branded test, see commit https://github.com/Azure/autorest.typescript/commit/d10ecbc75a1c2ff7529b16a3917796ca23f0ef25

### CODE REVIEW

File: packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+export interface TodoItemsCreateFormBodyParam {
+  body?: {
+    item: TodoItem;
+    attachments?: (
+      | TodoUrlAttachment
+      | string
+      | Uint8Array
+      | ReadableStream<Uint8Array>
+      | NodeJS.ReadableStream
+      | File
```
Comment (qiaozha) [inline packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts:83]: has the non-branded core already included this File type ?

Comment (qiaozha) [reply packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts:83]: I didn't see where it's imported.

Comment (MaryGao) [reply packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts:83]: File is not provided by core but by the environment.

File: packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  TodoItemsCreateJsonBodyParam &
+  RequestParameters;
+
+export interface TodoItemsCreateFormBodyParam {
+  body?: {
+    item: TodoItem;
+    attachments?: (
+      | TodoUrlAttachment
+      | string
+      | Uint8Array
```
Comment (MaryGao) [inline packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts:80]: With the [fix](https://github.com/Azure/autorest.typescript/pull/2258) we could generate the union of following for MFD. And @timovv would work on the sync between branding and non-branding.

@joheredi Do you think we could merge the codegen pr?
https://github.com/Azure/autorest.typescript/pull/2300/commits/d10ecbc75a1c2ff7529b16a3917796ca23f0ef25

**Pull request closed without merging**